modules/sqs/main.tf

```diff
@@ -139,6 +151,88 @@ resource "aws_sqs_queue" "metadata_queue" {
   visibility_timeout_seconds  = 900 # Set to the lambda max time
 }
 
+## s3-inventory-queue - get_current_archive_list lambda reads from this queue to get what it needs to pull in next
+## ==================================================================================================================
+resource "aws_sqs_queue" "s3_inventory_queue" {
+  ## OPTIONAL
+  name                        = "${var.prefix}-orca-s3-inventory-queue.fifo"
+  fifo_queue                  = true
+  content_based_deduplication = true
+  delay_seconds               = var.sqs_delay_time_seconds
+  max_message_size            = var.sqs_maximum_message_size
+  message_retention_seconds   = var.s3_inventory_queue_message_retention_time_seconds
+  tags                        = local.tags
+  policy                      = data.aws_iam_policy_document.s3_inventory_queue_policy.json
+  visibility_timeout_seconds  = var.orca_reconciliation_lambda_timeout
+  depends_on = [
+    aws_sqs_queue.s3_inventory_dlq
+  ]
+  redrive_policy = jsonencode({
+    deadLetterTargetArn = aws_sqs_queue.s3_inventory_dlq.arn,
+    maxReceiveCount     = 3 #number of times a consumer tries receiving a message from the queue without deleting it before being moved to DLQ.
+  })
+}
+
+# Dead-letter queue
+resource "aws_sqs_queue" "s3_inventory_dlq" {
+  name                       = "${var.prefix}-orca-s3-inventory-deadletter-queue.fifo"
```

Comment:
Per our discussion on FIFO and how a FIFO DLQ we think will break the issue where processing would stop because order must be maintained, will leave as FIFO for now until we see some real world use/testing.